Training app Heroku deployment failed

Hi,

While reading some information about Plone on your website, I clicked on "Deploy on Heroku" button from page http://plone.fr/essayer-plone

Then I filled the Heroku form to deploy the app in free instance, using Heroku Postgres and tag https://github.com/plone/heroku-buildpack-plone#5.0.8

Here is the result log

```
-----> Plone app detected
-----> Use build cache
       Cache empty, let's fix that
       Prime with Unified Installer cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
    0     0   152k      0  0:06:29  0:00:11  0:06:18  161k2 57.9M   42 24.3M    0     0   375k      0  0:02:38  0:01:06  0:01:32  366k 0   360k      0  0:02:44  0:01:17  0:01:27  213k9M   48 28.0M    0     0   361k      0  0:02:44  0:01:19  0:01:25  309k  0:02:25 --:--:--  254k
       Unpacking Installer
       Unpacking Installer cache
       Moving cache eggs to build directory
       Clean up Installer files
-----> Read BUILDOUT_CFG from env vars, or use default
       Using default: heroku.cfg
-----> Read BUILDOUT_VERBOSITY from env vars, or use default
       Use default buildout verbosity
-----> Read BOOTSTRAP_PY_URL from env vars, or use default
       Use default bootstrap.py: http://downloads.buildout.org/2/bootstrap.py
-----> Bootstrap buildout
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
curl: (6) Could not resolve host: L
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>301 Moved Permanently</title>
</head><body>
<h1>Moved Permanently</h1>
<p>The document has moved <a href="https://bootstrap.pypa.io/bootstrap-buildout.py">here</a>.</p>
<hr>
<address>Apache Server at downloads.buildout.org Port 80</address>
</body></html>
ez_setup.py is deprecated and when using it setuptools will be pinned to 33.1.1 since it's the last version that supports setuptools self upgrade/installation, check https://github.com/pypa/setuptools/issues/581 for more info; use pip to install setuptools
Downloading https://pypi.io/packages/source/s/setuptools/setuptools-33.1.1.zip
Extracting in /tmp/tmpQjH5Nq
Now working in /tmp/tmpQjH5Nq/setuptools-33.1.1
Building a Setuptools egg in /tmp/tmp2lH106
warning: no files found matching '*' under directory 'setuptools/_vendor'
/tmp/tmp2lH106/setuptools-33.1.1-py2.7.egg
/tmp/tmp2lH106/setuptools-33.1.1-py2.7.egg/pkg_resources/__init__.py:184: RuntimeWarning: You have iterated over the result of pkg_resources.parse_version. This is a legacy behavior which is inconsistent with the new version class introduced in setuptools 8.0. In most cases, conversion to a tuple is unnecessary. For comparison of versions, sort the Version instances directly. If you have another use case requiring the tuple, please file a bug with the setuptools project describing that need.
no previously-included directories found matching 'doc'
no previously-included directories found matching 'old-tutorial'
Creating directory '/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/parts'.
Creating directory '/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/develop-eggs'.
Generated script '/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/bin/buildout'.
-----> Run bin/buildout -c heroku.cfg 
Getting distribution for 'ZODB3==3.10.7'.
In file included from src/BTrees/BTreeModuleTemplate.c:394:0,
                 from src/BTrees/_IIBTree.c:34:
src/BTrees/BucketTemplate.c: In function ‘bucket_traverse’:
src/BTrees/BucketTemplate.c:1627:12: warning: variable ‘len’ set but not used [-Wunused-but-set-variable]
     int i, len;
            ^
In file included from src/BTrees/BTreeModuleTemplate.c:394:0,
                 from src/BTrees/_IFBTree.c:34:
src/BTrees/BucketTemplate.c: In function ‘bucket_traverse’:
src/BTrees/BucketTemplate.c:1627:12: warning: variable ‘len’ set but not used [-Wunused-but-set-variable]
     int i, len;
            ^
In file included from src/BTrees/BTreeModuleTemplate.c:394:0,
                 from src/BTrees/_fsBTree.c:83:
src/BTrees/BucketTemplate.c: In function ‘bucket_traverse’:
src/BTrees/BucketTemplate.c:1627:12: warning: variable ‘len’ set but not used [-Wunused-but-set-variable]
     int i, len;
            ^
In file included from src/BTrees/BTreeModuleTemplate.c:394:0,
                 from src/BTrees/_LLBTree.c:36:
src/BTrees/BucketTemplate.c: In function ‘bucket_traverse’:
src/BTrees/BucketTemplate.c:1627:12: warning: variable ‘len’ set but not used [-Wunused-but-set-variable]
     int i, len;
            ^
In file included from src/BTrees/BTreeModuleTemplate.c:394:0,
                 from src/BTrees/_LFBTree.c:36:
src/BTrees/BucketTemplate.c: In function ‘bucket_traverse’:
src/BTrees/BucketTemplate.c:1627:12: warning: variable ‘len’ set but not used [-Wunused-but-set-variable]
     int i, len;
            ^
Got ZODB3 3.10.7.
Getting distribution for 'zc.buildout==2.9.5'.
no previously-included directories found matching 'doc'
no previously-included directories found matching 'old-tutorial'
While:
  Installing.
  Getting section precompile.
  Initializing section precompile.
  Getting option precompile:eggs.
  Getting section instance.
  Initializing section instance.
  Installing recipe plone.recipe.zope2instance.
  Getting distribution for 'zc.buildout==2.9.5'.
An internal error occurred due to a bug in either zc.buildout or in a
recipe being used:
Traceback (most recent call last):
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/buildout.py", line 2123, in main
    getattr(buildout, command)(args)
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/buildout.py", line 680, in install
    [self[part]['recipe'] for part in install_parts]
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/buildout.py", line 1274, in __getitem__
    options._initialize()
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/buildout.py", line 1372, in _initialize
    self._dosub(k, v)
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/buildout.py", line 1427, in _dosub
    v = '$$'.join([self._sub(s, seen) for s in v.split('$$')])
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/buildout.py", line 1491, in _sub
    v = self.buildout[section].get(option, None, seen)
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/buildout.py", line 1274, in __getitem__
    options._initialize()
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/buildout.py", line 1382, in _initialize
    self.initialize()
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/buildout.py", line 1388, in initialize
    recipe_class = _install_and_load(reqs, 'zc.buildout', entry, buildout)
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/buildout.py", line 1338, in _install_and_load
    allow_hosts=buildout._allow_hosts
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/easy_install.py", line 913, in install
    return installer.install(specs, working_set)
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/easy_install.py", line 714, in install
    for dist in self._get_dist(req, ws):
  File "/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg/zc/buildout/easy_install.py", line 570, in _get_dist
    dist = self._env.best_match(requirement, ws)
  File "build/bdist.linux-x86_64/egg/pkg_resources/__init__.py", line 1111, in best_match
    dist = working_set.find(req)
  File "build/bdist.linux-x86_64/egg/pkg_resources/__init__.py", line 715, in find
    raise VersionConflict(dist, req)
VersionConflict: (zc.buildout 2.9.6 (/tmp/build_68c44fc594fdd32a012110952d32c268/collective-training-sandbox-8839db8/eggs/zc.buildout-2.9.6-py2.7.egg), Requirement.parse('zc.buildout==2.9.5'))
 !     Push rejected, failed to compile Plone app.
 !     Push failed
```

--- a/bin/compile
+++ b/bin/compile
@@ -67,7 +67,7 @@ if [ -f $ENV_DIR/BOOTSTRAP_PY_URL ]; then
     export "BOOTSTRAP_PY_URL=$(cat $ENV_DIR/BOOTSTRAP_PY_URL)"
     echo "Use custom bootstrap.py: ${BOOTSTRAP_PY_URL}" | indent
 else
-    export "BOOTSTRAP_PY_URL=http://downloads.buildout.org/2/bootstrap.py"
+    export "BOOTSTRAP_PY_URL=https://bootstrap.pypa.io/bootstrap-buildout.py"
     echo "Use default bootstrap.py: ${BOOTSTRAP_PY_URL}" | indent
 fi
 

